src/GoodsList.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+
+import './GoodsList.css';
+
+interface Props {
+  goodsFromServer: string[];
+}
+
+interface State {
+  goods: string[];
+  sortBy: string;
+  listOfLength: string[];
+  goodLength: string;
+}
+export class GoodsList extends React.Component<Props, State> {
+  state = {
+    goods: [...this.props.goodsFromServer],
+    // isReversed: false,
+    sortBy: 'initial',
+    listOfLength: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'],
+    goodLength: '1',
+  };
+
+  sortByAbc = () => {
+    this.setState({
+      sortBy: 'alphabetically',
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'sortByLength',
+    });
+  };
+
+  resetSort = () => {
+    this.setState({
+      goods: [...this.props.goodsFromServer],
+      sortBy: 'initial',
+      goodLength: '1',
+    });
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
```

Comment:
And that's when it hit me how to fix this typescript problem. Gotta try it myself! 😮